spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -72,16 +73,43 @@
 	 * Default {@link PubSubTemplate} constructor that uses {@link SimplePubSubMessageConverter}
 	 * to serialize and deserialize payloads.
 	 * @param publisherFactory the {@link com.google.cloud.pubsub.v1.Publisher} factory to
-	 * publish to topics
+	 * publish to topics. Only required if using publish methods
 	 * @param subscriberFactory the {@link com.google.cloud.pubsub.v1.Subscriber} factory
-	 * to subscribe to subscriptions
+	 * to subscribe to subscriptions. Only required if using subscriber or pull methods
 	 */
 	public PubSubTemplate(PublisherFactory publisherFactory,
 			SubscriberFactory subscriberFactory) {
 		this.publisherFactory = publisherFactory;
 		this.subscriberFactory = subscriberFactory;
-		this.subscriberStub = this.subscriberFactory.createSubscriberStub();
-		this.acknowledger = this.subscriberFactory.createAcknowledger();
+
+		if (this.subscriberFactory != null) {
+			this.subscriberStub = this.subscriberFactory.createSubscriberStub();
+			this.acknowledger = this.subscriberFactory.createAcknowledger();
+		}
+		else {
+			this.subscriberStub = null;
+			this.acknowledger = null;
+		}
+	}
+
+	/**
+	 * {@link PubSubTemplate} constructor that uses {@link SimplePubSubMessageConverter}
+	 * to serialize and deserialize payloads.
+	 * @param publisherFactory the {@link com.google.cloud.pubsub.v1.Publisher} factory to
+	 * publish to topics. Subscribe and pull methods are not supported
```

Comment:
There is a rule for code style that new public API must have it version reflected via `@since`: https://github.com/spring-projects/spring-framework/wiki/Code-Style#use-of-since